### PR TITLE
mysql-test, part 2

### PR DESCRIPTION
**Pull request by osheroff**
DON'T CLICK MERGE.  merge by hand into master instead.

- `DOUBLE PRECISION` as alias for `DOUBLE`
- `ASCII` as alias for `character set latin1`
- `SIGNED` integer flag
- BINARY is a valid character set name
- handle b'101', x'fkljlkj' style strings
- CREATE AGGREGATE FUNCTION
- unquoted, higher bit utf8
@zendesk/rules

### CODE REVIEW

File: src/main/antlr4/imports/mysql_idents.g4 [hunk 1]
```diff
@@ -38,13 +38,13 @@ SQL_COMMENT: '/*' ~'!' (.)*? '*/' -> skip;
 
 SQL_LINE_COMMENT: ('#' | '--') (~'\n')* ('\n' | EOF) -> skip;
 
-STRING_LITERAL: TICK ('\\\'' | '\'\'' | ~('\''))* TICK;
+STRING_LITERAL: [a-zA-Z]? TICK ('\\\'' | '\'\'' | ~('\''))* TICK;
```
Comment (vanchi-zendesk) [inline src/main/antlr4/imports/mysql_idents.g4]: isn't this a bit too relaxed? What else can come here other than b and x? maybe o for octal?

Comment (vanchi-zendesk) [inline src/main/antlr4/imports/mysql_idents.g4]: I am assuming x'blah' is hex and b'blah' is binary. Maybe I am wrong?

**Comment by vanchi-zendesk** [comment]
:+1:

File: src/main/antlr4/imports/mysql_idents.g4 (hunk 1 above)
Comment (osheroff) [inline src/main/antlr4/imports/mysql_idents.g4]: it's actually not liberal enough, https://dev.mysql.com/doc/refman/5.7/en/literals.html shows there's some stuff I'm still missing (`_utf8'string'`).  But ok, I'll see what's available.

**Comment by osheroff** [comment]
merged by hand.

**Pull request closed without merging**